[SPARK-4134] [core] Lower severity of some executor loss logs.

vanzin opened the pull request:
Don't log ERROR messages when executors are explicitly killed or when
the exit reason is not yet known.

Commit:
[SPARK-11789] [core] [yarn] Lower severity of some executor loss logs.

Don't log ERROR messaged when executors are explicitly killed or when
the exit reason is not yet known.

SparkQA commented:
**[Test build #46116 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46116/consoleFull)** for PR 9780 at commit [`2accff3`](https://github.com/apache/spark/commit/2accff3010b2c2bcb4bc2e47983ab9aaba89fecf).

SparkQA commented:
**[Test build #46116 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46116/consoleFull)** for PR 9780 at commit [`2accff3`](https://github.com/apache/spark/commit/2accff3010b2c2bcb4bc2e47983ab9aaba89fecf).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:\n  * `case class ExecutorKilled(reason: String) extends ExecutorLossReason(reason)`\n

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/46116/
Test FAILed.

Commit:
Oops deleted too much code.

Commit:
Restore comment.

SparkQA commented:
**[Test build #46127 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46127/consoleFull)** for PR 9780 at commit [`523ba69`](https://github.com/apache/spark/commit/523ba6980f25cee1590d658af8dceed255092a5d).

jerryshao commented on core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala:
```diff
@@ -492,6 +492,18 @@ private[spark] class TaskSchedulerImpl(
+    case ExecutorKilled(message) =>
+      logInfo(s"Executor $executorId on $hostPort killed by driver ($message)")
+    case _ =>
+      logError("Lost executor %s on %s: %s".format(executorId, hostPort, reason))
```
Is it better to unify to use string interpolation?

vanzin replied:
Sure.

jerryshao commented:
LGTM, indeed a nice improvement :).

SparkQA commented:
**[Test build #46127 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46127/consoleFull)** for PR 9780 at commit [`523ba69`](https://github.com/apache/spark/commit/523ba6980f25cee1590d658af8dceed255092a5d).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:\n  * `case class ExecutorKilled(reason: String) extends ExecutorLossReason(reason)`\n

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/46127/
Test PASSed.

Commit:
interpolate all logs.

SparkQA commented:
**[Test build #46161 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46161/consoleFull)** for PR 9780 at commit [`a645c11`](https://github.com/apache/spark/commit/a645c116fd1ed2b16e3d7807e51021b016a8d490).

SparkQA commented:
**[Test build #46161 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46161/consoleFull)** for PR 9780 at commit [`a645c11`](https://github.com/apache/spark/commit/a645c116fd1ed2b16e3d7807e51021b016a8d490).
 * This patch **fails PySpark unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:\n  * `case class ExecutorKilled(reason: String) extends ExecutorLossReason(reason)`\n

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/46161/
Test FAILed.

vanzin commented:
retest this please

SparkQA commented:
**[Test build #46260 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46260/consoleFull)** for PR 9780 at commit [`a645c11`](https://github.com/apache/spark/commit/a645c116fd1ed2b16e3d7807e51021b016a8d490).

SparkQA commented:
**[Test build #46260 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46260/consoleFull)** for PR 9780 at commit [`a645c11`](https://github.com/apache/spark/commit/a645c116fd1ed2b16e3d7807e51021b016a8d490).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:\n  * `case class ExecutorKilled(reason: String) extends ExecutorLossReason(reason)`\n

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/46260/
Test FAILed.

vanzin commented:
hmm unrelated failure (flaky test maybe, need to take a look). retest this please

SparkQA commented:
**[Test build #46278 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46278/consoleFull)** for PR 9780 at commit [`a645c11`](https://github.com/apache/spark/commit/a645c116fd1ed2b16e3d7807e51021b016a8d490).

SparkQA commented:
**[Test build #46278 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46278/consoleFull)** for PR 9780 at commit [`a645c11`](https://github.com/apache/spark/commit/a645c116fd1ed2b16e3d7807e51021b016a8d490).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:\n  * `case class ExecutorKilled(reason: String) extends ExecutorLossReason(reason)`\n

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/46278/
Test FAILed.

andrewor14 commented:
@vanzin does this overlap with https://issues.apache.org/jira/browse/SPARK-4134?

andrewor14 commented:
retest this please

vanzin commented:
@andrewor14 yes, I was actually looking for that bug instead of filing a new one, but couldn't find it. I'll update things.

SparkQA commented:
**[Test build #46342 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46342/consoleFull)** for PR 9780 at commit [`a645c11`](https://github.com/apache/spark/commit/a645c116fd1ed2b16e3d7807e51021b016a8d490).

SparkQA commented:
**[Test build #46342 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46342/consoleFull)** for PR 9780 at commit [`a645c11`](https://github.com/apache/spark/commit/a645c116fd1ed2b16e3d7807e51021b016a8d490).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds the following public classes _(experimental)_:\n  * `case class ExecutorKilled(reason: String) extends ExecutorLossReason(reason)`\n

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/46342/
Test PASSed.

Commit:
Move logic to define whether executor was killed to driver.

This has two advantages:
- standalone and mesos should also benefit from the less noisy logs.
- the driver can make different decisions based on whether executor
  was killed by idle timeout or because of a heartbeat failure.

Commit:
Merge branch 'master' into SPARK-11789

SparkQA commented:
**[Test build #46355 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46355/consoleFull)** for PR 9780 at commit [`df2cc7c`](https://github.com/apache/spark/commit/df2cc7c3e1d852ef164ae52a1cad2586dcb7f73e).

andrewor14 commented on core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala:
```diff
@@ -488,6 +493,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       doRequestTotalExecutors(
         numExistingExecutors + numPendingExecutors - executorsPendingToRemove.size)
     } else {
+
```
delete

andrewor14 commented on core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala:
```diff
@@ -474,12 +480,11 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       logWarning(s"Executor to kill $id does not exist!")
     }
 
-    // If an executor is already pending to be removed, do not kill it again (SPARK-9795)
```
need to keep this?

andrewor14 commented on yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala:
```diff
@@ -493,7 +493,7 @@ private[yarn] class YarnAllocator(
           logInfo(containerExitReason)
         }
-        ExecutorExited(0, exitCausedByApp, containerExitReason)
+        ExecutorExited(exitStatus, exitCausedByApp, containerExitReason)
```
this is unrelated right? just wanted to double check but this fix looks OK

vanzin replied:
yeah, unrelated. just noticed it while reading that code.

andrewor14 commented:
This change looks fine. Thanks for fixing it.

Commit:
Feedback.

SparkQA commented:
**[Test build #46360 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46360/consoleFull)** for PR 9780 at commit [`a717270`](https://github.com/apache/spark/commit/a71727091166d02e77e9a9fc64b56c86c023d313).

SparkQA commented:
**[Test build #46355 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46355/consoleFull)** for PR 9780 at commit [`df2cc7c`](https://github.com/apache/spark/commit/df2cc7c3e1d852ef164ae52a1cad2586dcb7f73e).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/46355/
Test FAILed.

SparkQA commented:
**[Test build #46360 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/46360/consoleFull)** for PR 9780 at commit [`a717270`](https://github.com/apache/spark/commit/a71727091166d02e77e9a9fc64b56c86c023d313).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/46360/
Test PASSed.

andrewor14 commented:
Merging into master 1.6.
